src/index.ts

```diff
@@ -74,253 +74,272 @@ const TYPED_MESSAGE_SCHEMA = {
 };
 
 /**
- * A collection of utility functions used for signing typed data
+ * Encodes an object by encoding and concatenating each of its members
+ *
+ * @param {string} primaryType - Root type
+ * @param {Object} data - Object to encode
+ * @param {Object} types - Type definitions
+ * @returns {Buffer} - Encoded representation of an object
  */
-const TypedDataUtils = {
-  /**
-   * Encodes an object by encoding and concatenating each of its members
-   *
-   * @param {string} primaryType - Root type
-   * @param {Object} data - Object to encode
-   * @param {Object} types - Type definitions
-   * @returns {Buffer} - Encoded representation of an object
-   */
-  encodeData(
-    primaryType: string,
-    data: Record<string, unknown>,
-    types: Record<string, MessageTypeProperty[]>,
-    useV4 = true,
-  ): Buffer {
-    const encodedTypes = ['bytes32'];
-    const encodedValues = [this.hashType(primaryType, types)];
-
-    if (useV4) {
-      const encodeField = (name, type, value) => {
-        if (types[type] !== undefined) {
-          return [
-            'bytes32',
-            value == null // eslint-disable-line no-eq-null
-              ? '0x0000000000000000000000000000000000000000000000000000000000000000'
-              : ethUtil.keccak(this.encodeData(type, value, types, useV4)),
-          ];
-        }
+function encodeData(
+  primaryType: string,
+  data: Record<string, unknown>,
+  types: Record<string, MessageTypeProperty[]>,
+  useV4 = true,
+): Buffer {
+  const encodedTypes = ['bytes32'];
+  const encodedValues: unknown[] = [hashType(primaryType, types)];
```

Comment:
This was another type change that was required. TypeScript assumed that this was supposed to be a `Buffer` array because the first entry was a `Buffer`. But it looks like it's just an array of unknown data, not just of buffers.